modules/options/src/main/scala/scala/build/internal/ScalaNightlyVersion.scala

```diff
@@ -0,0 +1,72 @@
+package scala.build.internal
+
+import com.github.plokhotnyuk.jsoniter_scala.core._
+import com.github.plokhotnyuk.jsoniter_scala.macros.JsonCodecMaker
+import coursier.cache.FileCache
+import coursier.util.{Artifact, Task}
+import dependency.ScalaVersion
+
+import scala.build.EitherCps.{either, value}
+import scala.build.Os
+import scala.build.errors.{BuildException, ScalaVersionError}
+import scala.concurrent.duration._
+import scala.util.control.NonFatal
+
+object ScalaNightlyVersion {
+
+  object Scala2Repo {
+    final case class ScalaVersion(name: String, lastModified: Long)
+    final case class ScalaVersionsMetaData(repo: String, children: List[ScalaVersion])
+
+    val codec: JsonValueCodec[ScalaVersionsMetaData] = JsonCodecMaker.make
+  }
+
+  private def downloadScala2RepoPage(cache: FileCache[Task]): Either[BuildException, Array[Byte]] =
+    either {
+      val scala2NightlyRepo =
+        s"https://scala-ci.typesafe.com/ui/api/v1/ui/nativeBrowser/scala-integration/org/scala-lang/scala-compiler"
```

Comment:
NIT No need of a string interpolator here:
```suggestion
        "https://scala-ci.typesafe.com/ui/api/v1/ui/nativeBrowser/scala-integration/org/scala-lang/scala-compiler"
```